crates/viewer/re_viewer/src/app.rs

```diff
@@ -1507,6 +1507,13 @@ impl App {
                         );
 
                         render_ctx.begin_frame();
+
+                        // Call begin_frame on caches only after starting the re_renderer frame
+                        // since we rely on the renderer frame index for some tracking of outdated resources.
+                        if let Some(store_hub) = &mut self.store_hub {
+                            store_hub.begin_frame_caches(render_ctx.active_frame_idx());
```

Comment:
I think we should only call this on the first pass of each frame, in case there are several. Otherwise the "used this frame" is wrong, since it really means "used this pass".

Reminder: an egui frame usually consist of a single pass, but sometimes two.

Reply:
yeah those "this frame" things tripped which is why I removed those used-this-frame booleans.
We can do the is_sizing_pass checks here to not call this in the first place, but it immediately makes me wonder what else is affected by this.
For instance re_renderer also ignores everything in sizing commands 😬

Reply:
I'll try this for this particular thing on Monday and test it a bit. I should also have a look at other caches like mesh cache and image decode cache and ensure they still work as expected!